### PR TITLE
keras/saving_utils: call losses.deserialize instead of losses.get in `compile_args_from_training_config`

### PR DESCRIPTION
**Pull request by suyash**
When loading a keras model saved in `h5` format, there is a problem with custom losses.

Reproduction: https://gist.github.com/suyash/d45f664f0a9d62fd4ab052b4040e6a16

Essentially, from the stack trace, the `keras.losses.get` function does not accept a `custom_objects` parameter, which makes loading a custom loss break.

This changes `compile_args_from_training_config` to call `losses.deserialize` instead of `losses.get`, similar to `optimizers.deserialize` a couple of lines above

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by pavithrasv** [comment]
Thank you for the fix, there is another PR that fixes the same issue  https://github.com/tensorflow/tensorflow/pull/33229 so i am going to close this one.
